common/djangoapps/entitlements/models.py

```diff
@@ -302,6 +304,45 @@ def get_active_entitlements_for_user(cls, user):
             enrollment_course_run=None
         ).select_related('user').select_related('enrollment_course_run')
 
+    @classmethod
+    def check_for_existing_entitlement_and_enroll(cls, user, course_run_key):
+        """
+        Looks at the User's existing entitlements to see if the user already has a Course Entitlement for the course run
+        provided in the course_key.  If the user does have an Entitlement with no run set, the User is enrolled in the mode
+        set in the Entitlement.
+
+        Note: This method assumes that eligibility to enroll in the course has already been checked.
+
+        Arguments:
+            user (User): The user that we are inspecting the entitlements for.
+            course_run_key (CourseKey): The course run Key.
+
+        Returns:
+            CourseEntitlement: The CourseEntitlement that was found and used and None otherwise.
+        """
+        # Check if the User has any Entitlements that are active
+        entitlements = CourseEntitlement.get_active_entitlements_for_user(user)
+        if len(entitlements) > 0:
+            course_uuid_str = get_course_uuid_for_course(course_run_key)
+            # Look for an entitlement for the current uuid
+            if course_uuid_str:
+                course_entitlement = CourseEntitlement.get_entitlement_if_active(
+                    user=user,
+                    course_uuid=uuid_tools.UUID(course_uuid_str)
```

Comment:
this uuid casting could be shifted into the get_course_uuid_for_course function.